clients/js/src/ChromaClient.ts

```diff
@@ -337,11 +383,377 @@ export class ChromaClient {
    * });
    * ```
    */
-  public async deleteCollection({
-    name,
-  }: DeleteCollectionParams): Promise<void> {
-    return await this.api
-      .deleteCollection(name, this.tenant, this.database, this.api.options)
-      .then(handleSuccess);
+  async deleteCollection({ name }: DeleteCollectionParams): Promise<void> {
+    await this.initPromise;
+    await this.api.deleteCollection(
+      name,
+      this.tenant,
+      this.database,
+      this.api.options,
+    );
+  }
+
+  /**
+   * Add items to the collection
+   * @param {Object} params - The parameters for the query.
+   * @param {ID | IDs} [params.ids] - IDs of the items to add.
+   * @param {Embedding | Embeddings} [params.embeddings] - Optional embeddings of the items to add.
+   * @param {Metadata | Metadatas} [params.metadatas] - Optional metadata of the items to add.
+   * @param {Document | Documents} [params.documents] - Optional documents of the items to add.
+   * @returns {Promise<AddResponse>} - The response from the API. True if successful.
+   *
+   * @example
+   * ```typescript
+   * const response = await client.addRecords(collection, {
+   *   ids: ["id1", "id2"],
+   *   embeddings: [[1, 2, 3], [4, 5, 6]],
+   *   metadatas: [{ "key": "value" }, { "key": "value" }],
+   *   documents: ["document1", "document2"]
+   * });
+   * ```
+   */
+  async addRecords(
+    collection: Collection,
```

Comment:
most (all?) methods that deal with documents/records only need the collection's ID, so maybe we could update the typing to reflect that?

```suggestion
    collection: Pick<Collection, "id">,
```

then people can avoid an API call to fetch full collection details if they really want to

Reply:
Yeah the one thing that we do regularly need is the embedding function so I think it'll beneficial for us to keep Collection as a kind of opaque object that they have to get from the API. My understanding is that there's work underway for the backend to store some representation of embedding function alongside the collection which I think will make this opaque Collection object make more sense.

@atroyn in case I've misunderstood anything here.

Reply:
personally, as a JS dev, I would assume any object returned from an idiomatic API client library is trivially serializable/de-serializable (`JSON.[stringify|parse]`)
as long as we can hold that assumption that sounds ok to me

Reply:
alternatively, if every single method only needs the collection ID, I do feel it would be more idiomatic to have methods accept `async addRecords(collection_id: string...)` instead (and store any necessary collection metadata on the client object)

Reply:
We will be able to hold that assumption with EF persist in the works

Reply:
@AlabasterAxe what do you think about using the collection ID for these methods?

Reply:
Okay so if I'm understanding what you proposed above, you're saying that the client would maintain a mapping from collection ids to embedding functions and then will just look that up whenever the user supplies a collection id?

One potential issue that could arise is that a user could attempt to call the `addRecords` method even if the embedding function had never been configured for a particular collection.

So hypothetically we could have the following situation:
```TypeScript
/* Run 1 */
const collection = client.createCollection({name: "test"});
console.log(collection.id) // prints 1234

/* Run 2 */
client.addRecords("1234", {/* record contents */}); // How do we know which embedding function to use 
                                                    // here if the client has not gotten the collection in this
                                                    // run of the script?
```

I'll defer to @atroyn here: Should we continue accepting collection objects in the record methods? or should we move to accepting collection ids in the API?

Reply:
discussed with @atroyn and we settled on keeping the methods accepting collection objects for now